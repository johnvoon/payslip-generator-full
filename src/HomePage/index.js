import React from "react";
import { Link } from "react-router";
import Helmet from "react-helmet";
import Footer from "components/Footer";

// eslint-disable-next-line no-unused-vars
const HomePage = props => {
  return (
    <div>
      <main>
        <Helmet
          title="Payslip Generator"
          meta={[
            {
              name: "description",
              content: "An application to generate CSV payslip data from CSV files."
            }
          ]}
        />
        <div
          className="jumbotron"
          style={{
            backgroundImage: "url(https://payslip-generator.surge.sh/images/calculator.jpg)"
          }}
        >
          <div className="container text-center">
            <Link to="/payslip-generator" className="btn btn-home">
              Go to Payslip Generator
            </Link>
          </div>
        </div>
        <div className="home-content container-fluid">
          <div className="lg-margin-top">
            <h1 className="text-center">Instructions</h1>
            <ol>
              <li>
                Go to {" "}
                <Link to="/payslip-generator">
                  Payslip Generator
                </Link>
              </li>
              <li>
                Upload a .csv file into the file upload field by drag-and-drop or by clicking it to browse your file system. Click "Reset Field" to clear the file upload field.
              </li>
              <li>
                Click "Generate Payslip Data".
              </li>
              <li>
                Enter an optional file name (defaults to "payslip.csv").
              </li>
              <li>
                Click "Download".
              </li>
            </ol>
          </div>
        </div>
      </main>
      <Footer />
    </div>
  );
};

export default HomePage;
