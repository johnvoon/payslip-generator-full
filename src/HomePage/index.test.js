import React from "react";
import HomePage from "./index";
import { shallow } from "enzyme";

describe("HomePage", () => {
  const renderedComponent = shallow(<HomePage />);

  it("should match snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
