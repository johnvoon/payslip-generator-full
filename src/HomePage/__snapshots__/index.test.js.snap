exports[`HomePage should match snapshot 1`] = `
<div>
  <main>
    <HelmetWrapper
      meta={
        Array [
          Object {
            "content": "An application to generate CSV payslip data from CSV files.",
            "name": "description",
          },
        ]
      }
      title="Payslip Generator" />
    <div
      className="jumbotron"
      style={
        Object {
          "backgroundImage": "url(https://payslip-generator.surge.sh/images/calculator.jpg)",
        }
      }>
      <div
        className="container text-center">
        <Link
          className="btn btn-home"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/payslip-generator">
          Go to Payslip Generator
        </Link>
      </div>
    </div>
    <div
      className="home-content container-fluid">
      <div
        className="lg-margin-top">
        <h1
          className="text-center">
          Instructions
        </h1>
        <ol>
          <li>
            Go to 
             
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/payslip-generator">
              Payslip Generator
            </Link>
          </li>
          <li>
            Upload a .csv file into the file upload field by drag-and-drop or by clicking it to browse your file system. Click \"Reset Field\" to clear the file upload field.
          </li>
          <li>
            Click \"Generate Payslip Data\".
          </li>
          <li>
            Enter an optional file name (defaults to \"payslip.csv\").
          </li>
          <li>
            Click \"Download\".
          </li>
        </ol>
      </div>
    </div>
  </main>
  <Footer />
</div>
`;
